### ISSUE TITLE
Unauthorized requests after token refresh for AJAX headers

### ISSUE THREAD

**Issue by raulgabor** [openseadragon/openseadragon#2550]
Hi,

I am using AJAX headers to request tiles and everything seems to work fine until the authorization token is refreshed and I receive the new one. I am using the setAjaxHeaders method for when the new token arrives and the set is made on the viewer so it'll propagate to all tiled images, respectively tiles. After the token refresh however, I receive some errors on the console about some GET request being unauthorized, even though the setAjaxHeaders has been called and the new token is set. The errors are for the level where I was when the token refreshed, but I can receive errors for a level up/down in some instances. After these, I get no more errors until the next token refresh. Do you have any idea or starting point on how I can fix this issue? Thank you!

![Capture](https://github.com/openseadragon/openseadragon/assets/74712989/8d722425-84ad-4d5b-ac0d-745f9d48dadc)

**Comment by iangilman** [comment]
I wonder if it has to do with tile loads already in process when the changeover happens?

**Comment by raulgabor** [comment]
@iangilman yes, this happens when I am navigating on the viewer and at the same time the token refreshes, but it also happens when I don’t navigate in the viewer and is at a certain zoom level; after the new token has been set and I start navigating the errors appear.

**Comment by iangilman** [comment]
Do you have the navigator up? I wonder if it's related to https://github.com/openseadragon/openseadragon/pull/2539, which hasn't been released yet.

**Comment by iangilman** [comment]
Yeah, that is interesting. Still, it seems the most likely thing to figure out first.

Sounds like it's something that would need fixing inside OSD, like a more aggressive propagation of the new headers. If you're interested in pursuing a fix, I can point you in the right direction.

**Comment by amudhanj26** [comment]
Hi @iangilman,
I find this issue interesting can i give it a try?
Trying to learn something new.
